osscameroon/jobsika
# feat: post a job offer - backend

Steps to verify:

  - move to the backend folder with cd ./backend
  - run make start-postgres to run an initialized database
  - run make run to launch the api
  - then run the following commands

1️⃣ POST Job offer
```shell 
curl -s -X POST '{ "email": "oss-cm@gmail.com", "company_name": "Gamesmania", "job_title": "Frontend Dev", "is_remote": true, "description": " Oss is hiring a Remote Go Backend Engineer \n Remote - We are looking for a backend engineer who can work 30+ hr/weekOur ideal candidate has:- 1+ years experience writing in Go: $70,000 — $120000/year", "how_to_apply": "", "apply_url": "", "apply_email_address": "oss-cm@gmail.com" }'
```

you should see something like this 👇

```json
{
    "id": 1,
    "createdat": "2023-01-29T09:48:59.4477866+01:00",
    "updatedat": "2023-01-29T09:48:59.4477866+01:00",
    "email": "oss-cm@gmail.com",
    "company_name": "Gamesmania",
    "job_title": "Frontend Dev",
    "is_remote": true,
    "description": " Oss is hiring a Remote Go Backend Engineer \n Remote - We are looking for a backend engineer who can work 30+ hr/weekOur ideal candidate has:- 1+ years experience writing in Go: $70,000 — $120000/year",
    "how_to_apply": "",
    "apply_url": "",
    "apply_email_address": "oss-cm@gmail.com"
}
```

2️⃣ Check mandatory field
You can check mandatory fields by submitting the same request without the mandatory fields.
Mandatory fields are: 

- Email
- Company Name
- Description
- Job Title
- Apply URL or Apply Email
